### PR TITLE
feat: Add cadvisor role

### PR DESCRIPTION
**Pull request by SuperQ**
Add a role to deploy https://github.com/google/cadvisor.

### CODE REVIEW

File: roles/cadvisor/defaults/main.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+---
+cadvisor_version: 0.47.2
+cadvisor_binary_local_dir: ""
+cadvisor_binary_url: "https://github.com/{{ _cadvisor_repo }}/releases/download/v{{ cadvisor_version }}/\
+                           cadvisor-v{{ cadvisor_version }}-linux-{{ go_arch }}"
```
Comment (gardar) [inline roles/cadvisor/defaults/main.yml:5]:
```suggestion
cadvisor_binary_url: "https://github.com/{{ _cadvisor_repo }}/releases/download/v{{ cadvisor_version }}/\
                      cadvisor-v{{ cadvisor_version }}-linux-{{ go_arch }}"
```

File: roles/cadvisor/meta/argument_specs.yml [hunk 2]
```diff
@@ -0,0 +1,49 @@
+---
+# yamllint disable rule:line-length
+argument_specs:
+  main:
+    short_description: "cAdvisor"
+    description:
+      - "Deploy L(cadvisor,https://github.com/google/cadvisor) using ansible"
+    author:
+      - "cAdvisor Authors"
```
Comment (gardar) [inline roles/cadvisor/meta/argument_specs.yml:9]: I think this field is for the role authors

File: roles/cadvisor/molecule/alternative/prepare.yml [hunk 3]
```diff
@@ -0,0 +1,25 @@
+---
+- name: Run local preparation
+  hosts: localhost
+  gather_facts: false
+  tasks:
+    - name: Download cadvisor binary to local folder
+      become: false
+      ansible.builtin.get_url:
+        url: "https://github.com/google/cadvisor/releases/download/v{{ cadvisor_version\
+          \ }}/cadvisor-v{{ cadvisor_version }}-linux-{{ go_arch }}"
```
Comment (gardar) [inline roles/cadvisor/molecule/alternative/prepare.yml:10]:
```suggestion
        url: "{{ cadvisor_binary_url }}"
```

**Review by gardar — CHANGES_REQUESTED** [review]

File: roles/cadvisor/meta/argument_specs.yml (hunk 2 above)
Comment (SuperQ) [reply roles/cadvisor/meta/argument_specs.yml:9]: Oh, ok.

File: roles/cadvisor/molecule/alternative/prepare.yml [hunk 4]
```diff
@@ -0,0 +1,24 @@
+---
+- name: Run local preparation
+  hosts: localhost
```
Comment (gardar) [inline roles/cadvisor/molecule/alternative/prepare.yml:3]: Perhaps we need something like

```suggestion
  hosts: localhost
  vars_files: 
    - ../defaults/main.yml
```

To make `cadvisor_binary_url` variable available

Comment (SuperQ) [reply roles/cadvisor/molecule/alternative/prepare.yml:3]: I put this back to the original pattern, which is what we have in all of the other role tests.

Comment (gardar) [reply roles/cadvisor/molecule/alternative/prepare.yml:3]: Right ok, we'll put this on the back burner for all the roles then

**Pull request merged**